core/space/services/services_sharing.go

```diff
@@ -225,12 +225,30 @@ func (s *Space) OpenSharedFile(ctx context.Context, hash, password, filename str
 	}, nil
 }
 
-func (s *Space) ShareFilesViaPublicKey(ctx context.Context, bucketName string, paths []string, pubkeys []crypto.PubKey) error {
-	err := s.tc.ShareFilesViaPublicKey(ctx, bucketName, paths, pubkeys)
+func (s *Space) ShareFilesViaPublicKey(ctx context.Context, paths []domain.FullPath, pubkeys []crypto.PubKey) error {
+	err := s.tc.ShareFilesViaPublicKey(ctx, paths, pubkeys)
 	if err != nil {
 		return err
 	}
+
+	for _, path := range paths {
+		if path.DbId == "" {
+			// TODO: fetch dbid of mirror bucket and set
+			// b, err := s.tc.GetDefaultBucket(ctx)
+			// if err != nil {
+			// 	return err
+			// }
+
+			// bs, err := s.tc.GetBucketSchema(ctx, b.Slug())
```

Comment:
@maurycy just wanted to check if there is any function like this coming with your PR before I go ahead and create one (i.e., external function to get the schema given the slug).

Reply:
@jsonsivar 

For `BucketSchema` (not `textile.Bucket`) there is:

```
func (tc *textileClient) findBucketInCollection(ctx context.Context, bucketSlug string) (*BucketSchema, error) {
```

Does that work for you?

Reply:
Yes it does, I made that function public for now and used that thank you.